contracts/BTCUtils.sol

```diff
@@ -152,24 +159,41 @@ library BTCUtils {
         return uint32(bytesToUint(_beLocktime));
     }
 
+    /// @notice          Extracts number of inputs as bytes
+    /// @dev             This is encoded as a VarInt, and errors for high values
+    /// @param _b        The tx to evaluate
+    /// @return          The number of inputs
+    function extractNumInputsBytes(bytes _b) public pure returns (bytes) {
+        return _b.slice(6, 1);
+    }
+
     /// @notice          Extracts number of inputs as integer
     /// @dev             This is encoded as a VarInt, and errors for high values
     /// @param _b        The tx to evaluate
     /// @return          The number of inputs
     function extractNumInputs(bytes _b) public pure returns (uint8) {
-        uint256 _n = bytesToUint(_b.slice(6, 1));
+        uint256 _n = bytesToUint(extractNumInputsBytes(_b));
         require(_n < 0xfd, "VarInts not supported");  // Error on VarInts
         return uint8(_n);
     }
 
-    /// @notice          Finds the location of the number of outpus
+    /// @notice          Finds the location of the number of outputs
     /// @dev             This depends on the number of inputs
     /// @param _b        The tx to evaluate
     /// @return          The index of the VarInt numTxOuts
     function findNumOutputs(bytes _b) public pure returns (uint256) {
         return 7 + (41 * extractNumInputs(_b));
```

Comment:
Can we break out this 7 + 41x thing into another function, so that if we already know the number of inputs we can call that version (and have it be cheaper)

Reply:
`function calcNumOutputsLocation(uint _ins) public pure returns (uint) {}`

Reply:
Ugh, I basically want infinite granularity on everything

Reply:
@prestwich  will open separate PR for this